src/vm/jithelpers.cpp

```diff
@@ -2396,7 +2396,7 @@ HCIMPL2(Object*, JIT_ChkCastClass_Portable, MethodTable* pTargetMT, Object* pObj
         if (pMT == pTargetMT)
             return pObject;
 
-        pMT = MethodTable::GetParentMethodTableOrIndirection(pMT);
```

Comment:
It is important to use `GetParentMethodTableOrIndirection` for performance here. Regular `GetParentMethod` has conditional checks that GetParentMethodTableOrIndirection avoids.

It is ok if you make it slower for Linux/ARM that you are trying to optimize, but it should not affect other platforms.

Reply:
@jkotas 

I have added `IndirectPointer`, which handles this scenario for all platforms except Linux/ARM. `GetParentMethodTableOrIndirection` for Linux/ARM is similar to `GetParentMethodTable`, for other platforms `GetParentMethodTableOrIndirection` is same as before.